Misc/NEWS.d/next/Core and Builtins/2023-06-24-10-34-27.gh-issue-105775.OqjoGV.rst

```diff
@@ -0,0 +1,4 @@
+Remove the LOAD_CLOSURE opcode and make it a pseudo-op instead. This enables
+
+* Superinstruction formation
+* Removal of checks for uninitialized variables
```

Comment:
I don't think this PR does enable either of these things. This is because `_PyCfg_OptimizeCodeUnit` is called before `_PyCfg_ConvertPseudoOps`, in `optimize_and_assemble_code_unit`. When we optimize, pseudo instructions are still present.

I think this PR is still a useful incremental step (there's no reason for `LOAD_CLOSURE` to exist as a distinct opcode in actual bytecode or in the eval loop), but actually enabling it to participate in these optimizations will require either a) refactoring how cell/local offsets are handled in the compiler so we don't need to track the distinction between `LOAD_FAST` and `LOAD_CLOSURE` through to `fix_cell_offsets`, and eliminating the `LOAD_CLOSURE` pseudo-op entirely, or b) adding explicit `LOAD_CLOSURE` support to some optimizations.

We could also experiment with trying to remove pseudo-ops before optimizing, but we'd need to audit for any dependence in the optimizer on pseudo-ops, and I think currently there'd also be a problem with `prepare_localsplus` (which calls `fix_cell_offsets`) being called after optimization, and needing `LOAD_CLOSURE`.